### PR TITLE
Kconfig: Expose gnrc/netif configurations

### PR DESCRIPTION
**Pull request by PeterKietzmann**
### Contribution description
This PR moves the configuration macros of `gnrc_netif` found [here](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/gnrc/netif/conf.h) to the `CONFIG_` namespace, and exposes them to Kconfig.

As agreed offline with @leandrolanzieri I did not include the default thread priority parameter because this should be defined by the platform, if not changed manually. We might need a better solution to this in future.

Furthermore, I did not include `GNRC_NETIF_NUMOF` because this won't be static anymore in the future.

### Testing procedure
- `examples/gnrc_networking` should still work as usual.
- You should be able to configure the module's parameters via `make menuconfig` in the test application.

### Issues/PRs references
Part of #12888

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
@leandrolanzieri thanks for the review. I addressed your comments.

**Comment by leandrolanzieri** [comment]
The message for b966ec232a44037e2fa9c8a48816303de8104bee  seems to be too long :confused:

File: examples/gnrc_minimal/Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ USEMODULE += gnrc_icmpv6_echo
 USEMODULE += prng_minstd
 
 CFLAGS += -DLOG_LEVEL=LOG_NONE  # disable log output
-CFLAGS += -DGNRC_PKTBUF_SIZE=512 -DGNRC_NETIF_IPV6_ADDRS_NUMOF=2 \
+CFLAGS += -DGNRC_PKTBUF_SIZE=512 -DCONF_GNRC_NETIF_IPV6_ADDRS_NUMOF=2 \
```
Comment (miri64) [inline examples/gnrc_minimal/Makefile]:
```suggestion
CFLAGS += -DGNRC_PKTBUF_SIZE=512 -DCONFIG_GNRC_NETIF_IPV6_ADDRS_NUMOF=2 \
```
?

Comment (PeterKietzmann) [reply examples/gnrc_minimal/Makefile]: Whops. Very good catch! Should be fixed here and in one other Makefile.

Comment (leandrolanzieri) [reply examples/gnrc_minimal/Makefile]: Nice catch

File: sys/net/gnrc/netif/Kconfig [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+menuconfig KCONFIG_MODULE_GNRC_NETIF
+    bool "GNRC network interface configurations"
```
Comment (leandrolanzieri) [inline sys/net/gnrc/netif/Kconfig]: To be consistent:
```suggestion
    bool "Configure GNRC network interface"
```

**Pull request merged**